segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/SegmentState.java

```diff
@@ -85,8 +89,9 @@ public static SegmentState deserialize(DataInputStream source) throws IOExceptio
             if (version == SERIALIZATION_VERSION) {
                 long segmentId = source.readLong();
                 String segmentName = source.readUTF();
+                long startOffset = source.readLong();
                 Map<UUID, Long> attributes = AttributeSerializer.deserialize(source);
-                return new SegmentState(segmentId, segmentName, attributes);
+                return new SegmentState(segmentId, segmentName, startOffset, attributes);
```

Comment:
Does this need to bump the Serializetion_version above?

Reply:
For this version we don't support backward compatibility as far as I know

Reply:
Isn't the version used to check for backwards compatibility? So shouldn't it be incremented precisely because this change is breaking backward compatibility?

Reply:
No, I meant that for our version of Pravega we don't care about backwards compatibility. 

Especially when Part 2 of this comes in and it will require Tier2 to be wiped out and started clean because of the new file layout.

Reply:
Fixed. Also filed #1956 to come up with a more elegant solution to this, since we'll need to use this pattern in a lot of places.